TeamCode/src/main/java/org/firstinspires/ftc/teamcode/auton/smc/StateMachineOpMode.java

```diff
@@ -0,0 +1,86 @@
+package org.firstinspires.ftc.teamcode.auton.smc;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.util.ElapsedTime;
+
+import org.firstinspires.ftc.teamcode.auton.smc.states.*;
+
+import java.util.HashMap;
+
+/* Autonomous State Machine */
+@Autonomous(name = "Autonomous State Machine")
+public class StateMachineOpMode extends OpMode {
+    private final ElapsedTime time = new ElapsedTime();
+
+    /* Holds the key to the entry of the current state in states HashMap. */
+    Class<? extends StateInterface> currentState;
+    DrivingState drivingState = new DrivingState();
+    ScoringState scoringState = new ScoringState();
+    IntakeState intakeState = new IntakeState();
+    IdleState idleState = new IdleState();
+    Class<? extends StateInterface> initialState = DrivingState.class;
+    public HashMap<Class<? extends StateInterface>, StateInterface> states = new HashMap<>();
```

Comment:
Try and see if you can make this into just an `ArrayList`, I'd be curious to see how it would look and if it would clean things up (not that they're dirty).

Reply:
What do you mean by this?